NEO-TAT/tat_flutter
# [Null-safety] Codebase Null-safety & deprecated apis upgrade

## Description

In this PR, I tried to

1. Cut all the codes related to the `null` type to let them clearly indicate its type.
2. Remove related packages that do not support `null-safety`, and use supported packages as a replacement.
3. Update the existing package to a `null-safety` supported version.
4. Remove some unnecessary files.
5. Remove the deprecated API and replace it with a new version of the API with the same features.

## TODO

### `lib/src`

- [x] connector
- [x] connector/core
- [x] connector/interceptors
- [x] file
- [x] model
- [x] model/course
- [x] model/course/table
- [x] model/ischool_plus
- [x] model/ntut
- [x] model/remote_config
- [x] model/seting
- [x] model/userdata
- [x] providers
- [x] task/course
- [x] task/course_oads
- [x] task/iplus
- [x] task/ntut
- [x] task/score
- [x] util

### `lib/ui`

- [x] pages/calendar
- [x] pages/course_detail
- [x] pages/course_table
- [x] pages/file_viewer
- [x] pages/log_console
- [x] pages/notification
- [x] pages/other
- [x] pages/password
- [x] pages/score
- [x] page/video_player
- [x] pages/web_view

## How to verify?

Since the `null-safety` upgrade has almost covered all the code for this project, the way to verify whether it is normal is to install the test version directly, and use the last released version for detailed functional testing and comparison.
